utils/src/tap_error_log.rs

```diff
@@ -0,0 +1,60 @@
+// Copyright (c) 2022 RBB S.r.l
+// opensource@mintlayer.org
+// SPDX-License-Identifier: MIT
+// Licensed under the MIT License;
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// https://github.com/mintlayer/mintlayer-core/blob/master/LICENSE
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::fmt::Display;
+
+pub trait LogError
+where
+    Self: Sized,
+{
+    fn log_err(self) -> Self;
+    fn log_err_pfx(self, prefix: &str) -> Self;
+    fn log_warn(self) -> Self;
+    fn log_warn_pfx(self, prefix: &str) -> Self;
+}
+
+impl<T, E: Display> LogError for Result<T, E> {
```

Comment:
I think that it is better to use `Debug` instead of `Display` for an error. It can be more verbose, but it is what we usually want anyway.

Reply:
Since we're using thiserror, it'll basically print the error that's defined there above every error. There's no really more information that we need unless our labels are bad.